andi/andi.py

```diff
@@ -330,3 +362,18 @@ def _ensure_can_provide_func(cont_or_call: Optional[ContainerOrCallableType]
     if isinstance(cont_or_call, Container):
         return cont_or_call.__contains__
     return cont_or_call
+
+
+def _raise_exception(class_or_func, arg_errors):
```

Comment:
A small suggestion: implement _exception_message function instead. It'd be easier to test, and it'd be more clear which exceptions the caller function raises.

Reply:
Done